rbac/providers/azure/aad_auth.py

```diff
@@ -82,3 +82,22 @@ def check_token(self, AUTH_TYPE):
                 "Accept": "application/json",
                 "Host": "graph.microsoft.com",
             }
+
+    def _check_token(self, AUTH_TYPE):
+        """check if Token exists and calls for then caches as a global variable 
+        if it does not"""
+        if self.graph_token is None or not self._time_left():
+            response = self.get_token(AUTH_TYPE)
+            self.token_creation_timestamp = dt.datetime.now()
+            self.graph_token = response.json()["access_token"]
+        if AUTH_TYPE.upper() == "SECRET":
+            return {"Authorization": self.graph_token}
+        elif AUTH_TYPE.upper() == "CERT":
+            return {"Authorization": self.graph_token, "Host": "graph.microsoft.com"}
+
+    def check_token_POST(self, AUTH_TYPE):
```

Comment:
This could be get_request_headers() and include the logic for known AUTH_TYPE logic (cert, secret). 
That way, the content becomes easier to test, and _check_token can be renamed to something describing the way it reloads an expired/absent access token (refresh_access_token, get_or_generate_access_token)... Whatever moves the code in the direction towards more self-description.